python/lsst/daf/butler/remote_butler/_factory.py

```diff
@@ -93,42 +98,61 @@ def create_factory_from_config(config: ButlerConfig) -> RemoteButlerFactory:
         if server_url_key in config:
             config[server_url_key] = replaceRoot(config[server_url_key], config.configDir)
         remote_config = RemoteButlerConfigModel.model_validate(config)
-        return RemoteButlerFactory.create_factory_for_url(str(remote_config.remote_butler.url))
+        return RemoteButlerFactory(remote_config.remote_butler, http_client=http_client)
 
     @staticmethod
-    def create_factory_for_url(server_url: str) -> RemoteButlerFactory:
-        return RemoteButlerFactory(server_url)
+    def create_factory_for_url(
+        server_url: str, http_client: httpx.Client | None = None
+    ) -> RemoteButlerFactory:
+        config = ButlerConfig(server_url)
+        return RemoteButlerFactory.create_factory_from_config(config, http_client=http_client)
 
-    def create_butler_for_access_token(
+    def _create_butler(
         self,
-        access_token: str,
         *,
-        butler_options: ButlerInstanceOptions | None = None,
-        use_disabled_datastore_cache: bool = True,
+        auth: RemoteButlerAuthenticationProvider,
+        butler_options: ButlerInstanceOptions | None,
+        enable_datastore_cache: bool = False,
     ) -> RemoteButler:
         if butler_options is None:
             butler_options = ButlerInstanceOptions()
         return RemoteButler(
             connection=RemoteButlerHttpConnection(
-                http_client=self.http_client, server_url=self.server_url, access_token=access_token
+                http_client=self.http_client, server_url=self.server_url, auth=auth
             ),
             defaults=RegistryDefaults.from_butler_instance_options(butler_options),
             cache=self._cache,
-            use_disabled_datastore_cache=use_disabled_datastore_cache,
+            use_disabled_datastore_cache=not enable_datastore_cache,
+        )
+
+    def create_butler_for_access_token(
+        self,
+        access_token: str,
+        *,
+        butler_options: ButlerInstanceOptions | None = None,
+        enable_datastore_cache: bool = False,
+    ) -> RemoteButler:
+        auth: RemoteButlerAuthenticationProvider
+        if self._config.authentication == "rubin_science_platform":
+            auth = RubinAuthenticationProvider(access_token)
+        else:
```

Comment:
```suggestion
        elif self._config.authentication == "cadc":
```
? and then an else to complain if it's not a recognized value? (also below).  Or are you hoping that it's impossible to get to the else because the model validation won't allow it? (and mypy will complain if you put in an else here because it will say both options were checked?)